package/MDAnalysis/coordinates/memory.py

```diff
@@ -0,0 +1,242 @@
+        self.n_atoms = self.coordinate_array.shape[self.format.find('a')]
+
+        kwargs.pop("n_atoms", None)
+        self.ts = self._Timestep(self.n_atoms, **kwargs)
+        self.ts.dt = dt
+        if dimensions is not None:
+            self.ts.dimensions = dimensions
+        # self.ts.frame = -1
+        self._read_next_timestep()
+
+    def set_array(self, coordinate_array, format='afc'):
+        """
+        Set underlying array in desired column format.
+
+        Parameters
+        ---------
+        coordinate_array : :class:`~numpy.ndarray object
+            The underlying array of coordinates
+        format
+            The order/shape of the return data array, corresponding
+            to (a)tom, (f)rame, (c)oordinates all six combinations
+            of 'a', 'f', 'c' are allowed ie "fac" - return array
+            where the shape is (frame, number of atoms,
+            coordinates)
+        """
+        self.coordinate_array = coordinate_array
+        self.format = format
+
+    def get_array(self):
+        """
+        Return underlying array.
+        """
+        return self.coordinate_array
+
+    def _reopen(self):
+        """Reset iteration to first frame"""
+        self.ts.frame = -1
+
+    def timeseries(self, asel=None, start=0, stop=-1, skip=1, format='afc'):
+        """Return a subset of coordinate data for an AtomGroup in desired
+        column format. If no selection is given, it will return a view of
+        the underlying array, while a copy is returned otherwise.
+
+        Parameters
+        ---------
+        asel : :class:`~MDAnalysis.core.AtomGroup.AtomGroup` object
+            Atom selection
+        start, stop, skip : int
+            range of trajectory to access, start and stop are inclusive
+        format : str
+            the order/shape of the return data array, corresponding
+            to (a)tom, (f)rame, (c)oordinates all six combinations
+            of 'a', 'f', 'c' are allowed ie "fac" - return array
+            where the shape is (frame, number of atoms,
+            coordinates)
+        """
+        array = self.get_array()
+        if format == self.format:
+            pass
+        elif format[0] == self.format[0]:
+            array = np.swapaxes(array, 1, 2)
+        elif format[1] == self.format[1]:
+            array = np.swapaxes(array, 0, 2)
+        elif format[2] == self.format[2]:
+            array = np.swapaxes(array, 0, 1)
+        elif self.format[1] == format[0]:
+            array = np.swapaxes(array, 1, 0)
+            array = np.swapaxes(array, 1, 2)
+        elif self.format[2] == format[0]:
+            array = np.swapaxes(array, 2, 0)
+            array = np.swapaxes(array, 1, 2)
+
+        a_index = format.find('a')
+        f_index = format.find('f')
+        stop_index = stop+1
+        if stop_index == 0:
+            stop_index = None
+        basic_slice = ([slice(None)]*(f_index) +
```

Comment:
Extreme nitpicking, PEP8 says that there should be spaces around operators, so there should be one space on each side of the `*`. Other occurrences bellow.